### PR TITLE
Add preprocessor option handling

### PR DESCRIPTION
**Pull request by zaucker**
Get preprocessor options from command line and config file (web).

### CODE REVIEW

File: lib/Agrammon/UI/CommandLine.pm6 [hunk 1]
```diff
@@ -76,7 +78,9 @@ sub latex (IO::Path $path, $tech-file) is export {
     $path.dirname ~~ / .* '/' (.+) /;
     my $model-name = ~$0;
     my $model = timed "Load $module of $module-path from cache", {
-        load-model-using-cache( $*HOME.add('.agrammon'), $module-path, $module)
+        my %preprocessor-options = set($variants.split(","));
+        dd %preprocessor-options;
```
Comment (jnthn) [inline lib/Agrammon/UI/CommandLine.pm6:82]: Debug leftover

File: lib/Agrammon/UI/CommandLine.pm6 [hunk 2]
```diff
@@ -101,18 +105,21 @@ sub USAGE() is export {
...
     die "ERROR: dump expects a .nhd file" unless $path.extension eq 'nhd';
 
     my $module-path = $path.parent;
     my $module-file = $path.basename;
     my $module      = $path.extension('').basename;
 
-    my $model = timed "load $module", { load-model-using-cache($*HOME.add('.agrammon'), $module-path, $module) };
+    my $model = timed "load $module", {
+        my %preprocessor-options = set($variants.split(","));
+       load-model-using-cache($*HOME.add('.agrammon'), $module-path, $module, %preprocessor-options)
```
Comment (jnthn) [inline lib/Agrammon/UI/CommandLine.pm6:117]: Something looks a space off with the indentation here

File: lib/Agrammon/UI/CommandLine.pm6 [hunk 3]
```diff
@@ -197,7 +206,8 @@ sub web(Str $cfg-filename, Str $model-filename, Str $tech-file?) is export {
     }
 
     my $model = timed "Load model from $module-path/$module.nhd", {
-        load-model-using-cache($*HOME.add('.agrammon'), $module-path, $module)
+        my %preprocessor-options = set($variants.split(","));
```
Comment (jnthn) [inline lib/Agrammon/UI/CommandLine.pm6:209]: I do wonder if this line is repeated enough to be worth pulling the code out into a sub.

File: lib/Agrammon/UI/CommandLine.pm6 (hunk 1 above)
Comment (zaucker) [reply lib/Agrammon/UI/CommandLine.pm6:82]: Removed

File: lib/Agrammon/UI/CommandLine.pm6 (hunk 2 above)
Comment (zaucker) [reply lib/Agrammon/UI/CommandLine.pm6:117]: Fixed.

File: lib/Agrammon/UI/CommandLine.pm6 (hunk 3 above)
Comment (zaucker) [reply lib/Agrammon/UI/CommandLine.pm6:209]: Done.

**Review by jnthn — APPROVED** [review]

**Pull request merged**